File: CS50/sentimental-cash/cash.py
from cs50 import get_float

while True:
    change = get_float("Change: ")
    if change > 0:
        break

number = 0

while change >= 0.25:
    change = round(change - 0.25, 2)
    number += 1

while change >= 0.100:
    change = round(change - 0.10, 2)
    number += 1

while change >= 0.050:
    change = round(change - 0.05, 2)
    number += 1

while change >= 0.010:
    change = round(change - 0.01, 2)
    number += 1

print(number)

File: CS50/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int change_ammount();
int coin_numbers(int change);



int main(void)
{
    int c = change_ammount();       // Getting change ammount
    int n = coin_numbers(c);        // Finding coin numbers
    printf("%d\n", n);
}



// Getting change ammount
int change_ammount()
{
    int change = 0;

    do{
        change = get_int("Change owed: ");
    }
    while(change <= 0);

    return change;
}

// Determining number of coins
int coin_numbers(int change)
{
    int number = 0;

    while(change >= 25)
    {
        change = change -25;
        number++;
    }
    while(change >= 10)
    {
        change = change -10;
        number++;
    }
    while(change >= 5)
    {
        change = change -5;
        number++;
    }
    while(change >= 1)
    {
        change = change -1;
        number++;
    }

    return number;
}
